### PR TITLE
CMD parsing

### PR DESCRIPTION
**Pull request by connerCpowell**

### CODE REVIEW

File: ppB.py [hunk 1]
```diff
@@ -0,0 +1,393 @@
...
+                        help="CSV Directory: Location to save MS extraction .csv files \n")
+
+    parser.add_argument("-e",
+                        "--EXPRdir",
+                        action="store",
+                        dest="sdir2",
+                        help="EXPR Directory: Location to save the .expr files for alignment scripts \n")
+    args = parser.parse_args()
+
+    dirc = args.dirc
```
Comment (DailyDreaming) [inline ppB.py]: Probably don't need:

```
    dirc = args.dirc
    sp = args.name
    points = args.points
    scans = args.scans
    percent = args.threshold
    nin = args.ion
    sdir = args.sdir
    sdir2 = args.sdir2
```

It's less clutter to just keep all of these as `args.whatever`.

File: ppB.py [hunk 2]
```diff
@@ -0,0 +1,393 @@
...
+from pyms.Experiment.Class import Experiment
+from pyms.Experiment.IO import store_expr
+import itertools
+import fnmatch
+import os
+from pyms.Noise.Analysis import window_analyzer
+import csv
+import argparse
+
+import re
```
Comment (DailyDreaming) [inline ppB.py]: You seem to have a bunch of unused imports:

```
import re
import sys
from pyms.Peak.Class import Peak
```

File: ppB.py [hunk 3]
```diff
@@ -0,0 +1,393 @@
...
+            head = ['Area', 'RTs', 35.0, 36.0, 37.0, 38.0, 39.0, 40.0, 41.0, 42.0, 43.0, 44.0, 45.0, 46.0, 47.0, 48.0,
```
Comment (DailyDreaming) [inline ppB.py]: This could be written as `head = ['Area', 'RTs'] + [float(i) for i in range(35, 221)]`.

File: ppB.py [hunk 4]
```diff
@@ -0,0 +1,393 @@
...
+                # print('area:', area)
+                peak.set_area(area)
+                ms = peak.get_mass_spectrum()
+                # print("Peaks rt: ", peak.get_rt())
+                # print("Peaks ms_list: ", ms.mass_list)
+                # print("Peaks ms_spec: ", list(ms.mass_spec))
+                p_rt = peak.get_rt()
+                its = []
+                items = list(ms.mass_spec)
+                for i in items:
```
Comment (DailyDreaming) [inline ppB.py]: As an outsider:

```
                for i in items:
                    x = float(i)
                    its.append(x)
```

Code with ambiguous naming is more difficult to read.  It's a good habit, if you want other people to quickly pick up your code and easily make fixes or develop on it to make more descriptive names.

File: ppB.py [hunk 5]
```diff
@@ -0,0 +1,393 @@
...
+    savePath = sdir
+    ms_data_files = []
+
+    print("len pp_im", len(list(pp_im)))
+    print("len noise", len(noise))
+    print("len name", len(name), name)
+
+    for im, n, na in itertools.izip(list(pp_im), noise, name):
+
+        ms_data = []
```
Comment (DailyDreaming) [inline ppB.py]: This is unused.

File: ppB.py [hunk 6]
```diff
@@ -0,0 +1,393 @@
...
+            # print('ms', name1)
+
+            pp = open(name2, "w+")
+            for ratios, p, a in zip(ratio_set, peaks, areas):
+                # print('ratios=', ratios)
+                rs = sorted(ratios, key=lambda t: t[1], reverse=True)[:10]
+
+                # print('rs=', rs)
+                ll = 'Name:', p, 'Area-', a
+                ll = str(ll).replace("'", "").replace(",", "").replace("(", "").replace(")", "")
```
Comment (DailyDreaming) [inline ppB.py]: These are repeated and I'd recommend making them into functions.  This code:

```
            pp = open(name2, "w+")
            for ratios, p, a in zip(ratio_set, peaks, areas):
                # print('ratios=', ratios)
                rs = sorted(ratios, key=lambda t: t[1], reverse=True)[:10]

                # print('rs=', rs)
                ll = 'Name:', p, 'Area-', a
                ll = str(ll).replace("'", "").replace(",", "").replace("(", "").replace(")", "")
                # print('ll=', ll)
                pp.write(ll + "\n")

                nn = 'Num Peaks:', 10
                nn = str(nn).replace("'", "").replace(",", "").replace("(", "").replace(")", "")
                # print('nn=', nn)
                pp.write(nn + "\n")
                for i in rs:
                    ss = str(i).replace('[', '').replace(']', '').replace(',', '')
                    # print('ss=', ss)
                    pp.write(ss + "\n")
                # print('\n')
                pp.write("\n")
            pp.close()
```

I would rewrite as something like:

```
            def replace_n_write(ll, pp, chars):
                ll = str(ll)
                for c in chars:
                    ll = ll.replace(c, '')
                pp.write(ll + "\n")

            with open(name2, "w+") as pp:
                for ratios, p, a in zip(ratio_set, peaks, areas):
                    replace_n_write(ll=('Name:', p, 'Area-', a), pp=pp, chars="',()")
                    replace_n_write(ll=('Num Peaks:', 10), pp=pp, chars="',()")

                    for i in sorted(ratios, key=lambda t: t[1], reverse=True)[:10]:
                        replace_n_write(ll=i, pp=pp, chars="[],")
                    pp.write("\n")
```

Also, always try to use the `with open() as f` context manager convention, it's a good habit.

File: ppB.py [hunk 7]
```diff
@@ -0,0 +1,393 @@
...
+        areas = []
+        print('n=', n)
+        print("-------------11------------------------------------")
+        with open(n, 'r') as f:
+            next(f)
+            for line in f:
+                sline = line.split(',')
+                a = sline[0]
+                p = sline[1]
+                # p2 = sline[2]
```
Comment (DailyDreaming) [inline ppB.py:179]: Lots of unused comments could probably be cleaned up.

**Review by DailyDreaming — COMMENTED** [review]
Overall looks decent.  There's a lot of clean up and cruft like unused imports and commented out code that I'd delete to make it cleaner.  There are a ton of print statements that it might be worthwhile to delete or comment out or just make one option to turn them on or off.

I'd also recommend porting to python3 if you can, and maybe running `flake8` over this if you want an idea of some of the naming conventions.

**Comment by DailyDreaming** [comment]
Oh and tests should definitely be added.  That's actually probably the most important thing because it just makes sure the code behaves as you expect it to.

File: ppB.py (hunk 1 above)
Comment (connerCpowell) [reply ppB.py]: Replaced and removed.

File: ppB.py (hunk 2 above)
Comment (connerCpowell) [reply ppB.py]: Ill remove the unused imports.

Peak is used twice though, on line 136 & 137:
`peak.set_area(area)`
`ms = peak.get_mass_spectrum()`

File: ppB.py (hunk 3 above)
Comment (connerCpowell) [reply ppB.py]: Thank you!

File: ppB.py (hunk 4 above)
Comment (connerCpowell) [reply ppB.py]: Good call!
Renamed.

File: ppB.py (hunk 5 above)
Comment (connerCpowell) [reply ppB.py]: Removed ms_data.

File: ppB.py (hunk 6 above)
Comment (connerCpowell) [reply ppB.py]: `replace_n_write(ll=('Name:', p, 'Area-', a), pp=pp, chars="',()")
`
Is it needed to explicitly state, 'll=...'  or is the position of the variable enough for the script & readers to understand?

If it is for clarity's sake I understand as well. Just curious.

**Pull request merged**